### DISCUSSION TITLE
Samsung Galaxy S22 flickering

### DISCUSSION THREAD

**Discussion by lockeftw** [mapsforge/vtm#1128] [General]
Dear Mapsforge developer, 
I haveing following issue with Mobile Device Galaxy s22 ultra on Android 14 /13: 
Loading an offline works. But displaying this app works accept for the background , all Texts and pathes seems to work. 
The following links are save to my personal cloud to see some screen recordings of  apps that using mapsforge. 
https://cloud.datapiccante.de/s/ayswyNGXo2GnGe3
https://cloud.datapiccante.de/s/Cd4bCoGm78Yc2AZ

best regards Marc

**Comment by devemux86** [comment]
It seems you are using the [VTM OpenGL library](https://github.com/mapsforge/vtm).

We cannot know your implementation, so it is better to study the VTM samples,
like the [GettingStarted](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/GettingStarted.java) and [MapsforgeActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/MapsforgeActivity.java).

Do not create multiple map views and with fragments recreate the map view like the above examples.

**Comment by lockeftw** [comment]
I am sorry for my bad english. 

I am not the developer of these apps. 
Currently i have tried the apps c:geo, PocketMaps (first vid)  and GcDroid (second Vid) and on all of these apps the behavior is the same like in the scrrenrecordings.
 c:geo and PocketMaps are mentioned on https://github.com/mapsforge/mapsforge/blob/master/docs/Mapsforge-Applications.md.
Is this something Samsung Galaxy s22 ultra related ? .

**Comment by devemux86** [comment]
You can also try Cruiser: https://github.com/devemux86/cruiser

---

About the other apps it is better to also report in their forums.

**Comment by lockeftw** [comment]
I have downloaded and tested Cruiser with screenrecording: https://cloud.datapiccante.de/s/mBABBCotTQwsjJC .
Same behavior as the other apps.

**Reply by devemux86** [reply]
@lockeftw thanks for the test.

This is strange as Cruiser is used by many users and there is no such report.

Can you test on another device?
(to see if there is an OpenGL problem on that device)

**Reply by lockeftw** [reply]
I just have another Samsung galaxy s22 ultra and an old samsung galaxy s20 fe

**Comment by tqu-dev** [comment]
Are you seeing the same issue with Cartograph Maps 3 (https://play.google.com/store/apps/details?id=eu.cartograph3)?

**Reply by lockeftw** [reply]
Yes its same issue with Cartograph Maps 3.

**Reply by tqu-dev** [reply]
On a S23 Ultra it works fine with the apps you mentioned. Maybe it is indeed a S22 specific problem.

**Reply by lockeftw** [reply]
@tqu-dev In that case: i think i couldn't be alone with that problem XD.

**Comment by lockeftw** [comment]
I am not in to OpenGL that much, but i found a stackoverflow question, which concluses thats a device specific problem: 
https://stackoverflow.com/questions/75471741/opengl-es-1-fog-and-texture-transparency 
I hope there is a solution for this... looking forward.

**Comment by moving-bits** [comment]
This might indeed be a device-specific issue. Have tested it against a Samsung S22 and a Motorola device, with identical Cruiser version 4.0.2, OpenAndroMaps Berlin map and "Biker" theme:

(all artifacts "provoked" by simply panning the map slightly)

## a) background coloring of a specific area

![Screenshot_20240525_104707_Cruiser](https://github.com/mapsforge/vtm/assets/3754370/2acfb7bb-cc30-4a3a-993f-a2d52f8042fc).![Screenshot_20240525_104717_Cruiser](https://github.com/mapsforge/vtm/assets/3754370/723d72c6-c5fc-488b-a212-7231e1389c2a)

## b) background coloring of whole map tiles
![Screenshot_20240525_140147_Cruiser](https://github.com/mapsforge/vtm/assets/3754370/5148ca30-2ffd-4928-b044-9dde8358c090).![Screenshot_20240525_140154_Cruiser](https://github.com/mapsforge/vtm/assets/3754370/0705475a-e44c-4bfd-8858-ffbb75e390ac)

All screenshots are taken with the Samsung device. With the Motorola device everything seems to be fine (in my very short test).

**Reply by devemux86** [reply]
@moving-bits thanks for the tests!

**Reply by moving-bits** [reply]
Addtional info: I'm also able to get such effects when using Elevate:Elevate theme, but, again, on S22 only.

Also seems to be zoom level-specific: 
- When zoomed out, no problems occur. 
- On zoom level 14, greenish background of a certain graveyard toogles, on zoom level 15 it's stable, whereas on zoom level 16 it's cut in half vertically (right half with greenish background, the other half with white background).
- For the zoom level 14-issue it seems to depend on whether the tomb stones are being displayed or not.

![image](https://github.com/mapsforge/vtm/assets/3754370/f4284638-8baf-493b-ba69-fef353695019)

![image](https://github.com/mapsforge/vtm/assets/3754370/af3140a1-558b-491a-ae97-e04ee1114fb6).![image](https://github.com/mapsforge/vtm/assets/3754370/a89be1a6-741e-4eca-8415-46311bd446e5)

**Reply by muromir** [reply]
Hello,
from my perspective VTM maps on Samsung S22 are very difficult to use. I am facing the same issue as above mentioned. You can find below 3 screenshots showing blue areas which can be identified as water in random places, it depends on zoom level also forest is not visible . Screenshots taken from Cruiser 4.0.2.  
![Screenshot_20240611_131220_Cruiser](https://github.com/mapsforge/vtm/assets/104918108/dfa25f5b-5fad-4732-a91e-45b4cdb666fe)
![Screenshot_20240611_131300_Cruiser](https://github.com/mapsforge/vtm/assets/104918108/952e8f02-fe85-4ce7-aae9-a160c8bcd837)
![Screenshot_20240611_131315_Cruiser](https://github.com/mapsforge/vtm/assets/104918108/128aa5cd-d8e9-4981-918c-7d5b97ec7ce3)

**Comment by devemux86** [comment]
Because of the various OpenGL problems, Cruiser will return to Mapsforge:

I already implemented **map rotation** in Mapsforge:
- https://github.com/mapsforge/mapsforge/issues/7#issuecomment-2147721066

**Comment by DolomitesTom** [comment]
Only because of a specific problem on a specific Samsung S22 mobile phone was this terrible decision made to stop using openGL? I can't believe it 🙈😱.
